integration/src/test/java/com/fitgoal/NotificationServiceIntegrationTest.java

```diff
@@ -0,0 +1,106 @@
+package com.fitgoal;
+
+import javax.ws.rs.client.Entity;
+import javax.ws.rs.core.Response;
+
+import org.junit.Before;
+import org.junit.Test;
+
+import com.fitgoal.api.NotificationService;
+import com.fitgoal.api.domain.Recipient;
+import com.fitgoal.api.domain.UserVerification;
+
+import okhttp3.HttpUrl;
+
+import static org.assertj.core.api.Assertions.assertThatIllegalArgumentException;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class NotificationServiceIntegrationTest extends AbstractIntegrationTest {
+
+    private static NotificationService notificationService;
+
+    @Before
+    public void setup() {
+        notificationService = new NotificationClient(okHttpClient, HttpUrl.get(resourcePath));
+    }
+
+    @Test
+    public void registerTest() {
+        UserVerification userVerification = new UserVerification("test@test.com", "verification_link");
+        Response response = APP_EXTENSION.client()
+                .target(resourcePath)
+                .path("/register")
+                .request()
+                .post(Entity.json(userVerification));
+
+        assertThat(response).isNotNull();
+        assertThat(response.getStatus()).isEqualTo(204);
+    }
+
+    @Test
+    public void registerIncorrectEmailTest() {
+        UserVerification userVerification = new UserVerification("testtest.com", "verification_link");
+        assertThatIllegalArgumentException().isThrownBy(
```

Comment:
Minor but: I wouldn't mix testing instruments here (simple HTTP client and your own-written API client), I would cover all test cases with simple HTTP clients and all exceptional cases cover in `http-client` modules with unit tests.

Reply:
Done